cpp/include/algo_types.h

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Note: use this header for data structures, enums that need to be exposed to the API */
+
+#pragma once
+
+typedef enum {CUGRAPH_WEAK = 0, CUGRAPH_STRONG} cugraph_connect_t;
```

Comment:
Should `cugraph_connect_t ` be just `cugraph_cc_t `? `connect` looks ambiguous to me, or maybe   I've been looking at snmg stuff too much :)

Reply:
"connect" is _more_ ambiguous than "cc"? How?

Reply:
`connect` is very often used in communications/interconnect area, this could conflict if we use `connect` for unrelated MG things. eg. once could think of `cugraph_connect_t ` as `typedef enum {CUGRAPH_PCIE = 0, CUGRAPH_NVLINK} cugraph_connect_t;` 
`cc` was just a random example there might be something better.

Reply:
Ok. `cugraph_cc`_t it is then.